scripts/west_commands/runners/sctool.py

```diff
@@ -59,22 +65,42 @@ def _flash(self):
         if not build_conf['CONFIG_SOC_RISCV_TELINK_W91']:
             print('only Telink W91 chip is supported!')
             exit()
+        # get flash size
+        flash_size = hex(build_conf['CONFIG_FLASH_SIZE'] * 1024)
+        # get flash base address
+        if (self.address != self._default_addr()): # if user defined use it
+            print(f'User defined flash base address: {self.address}')
+            flash_base = self.address
+        else:
+            # calculate flash base address from configuration
+            if (build_conf.getboolean('CONFIG_MCUBOOT')): # mcuboot
+                print('MCUboot app!')    
+                flash_base = hex(build_conf['CONFIG_FLASH_BASE_ADDRESS'])
+            elif (build_conf.getboolean('CONFIG_BOOTLOADER_MCUBOOT')): # MCU boot app
+                print('MCUboot bootloader app!')
+                flash_base = hex(build_conf['CONFIG_FLASH_BASE_ADDRESS'] +
+                                 build_conf['CONFIG_TELINK_W91_SLOT0_PARTITION_ADDR'])
+            else: # standalone app
+                print('Standalone app!')
+                flash_base = hex(build_conf['CONFIG_FLASH_BASE_ADDRESS'])
+        
+        n22_partition_size = hex(build_conf['CONFIG_TELINK_W91_N22_PARTITION_SIZE'] * 1024)
+        n22_partition_addr = hex(build_conf['CONFIG_TELINK_W91_N22_PARTITION_ADDR'] + int(flash_base, 16))
         # download RAM
-        pre_erase = subprocess.Popen(['./sctool', '-p', self.usb_port, '-da', 'da/da.ram.bin', '--before', 'hw_reset', '--after', 'no_reset', '-b', '115200', 'upload_da'], cwd=self.sctool_path)
+        pre_erase = subprocess.Popen(['./sctool', '-p', self.usb_port, '-da', 'da/da.ram.bin', '-b', '2000000', '--before', 'hw_reset', '--after', 'no_reset', 'upload_da'], cwd=self.sctool_path)
```

Comment:
lets use variable of UART baud-rate instead of hardcoded

Reply:
done